plugins/woocommerce-blocks/tests/e2e/tests/product-gallery/product-gallery.block_theme.spec.ts

```diff
@@ -606,26 +624,21 @@ test.describe( `${ blockData.name }`, () => {
 		page,
 	} ) => {
 		await pageObject.addProductGalleryBlock( { cleanContent: true } );
-		await editor.saveSiteEditorEntities();
+		await editor.saveSiteEditorEntities( {
+			isOnlyCurrentEntityDirty: true,
+		} );
 
+		// Switch to the Index template.
 		await page.getByLabel( 'Open Navigation' ).click();
-		const navigationSidebar = page.getByLabel( 'Navigation' );
-		const navigationBackButton = navigationSidebar.getByLabel( 'Back' );
-		await expect( navigationBackButton ).toBeVisible();
-		await navigationSidebar.getByLabel( 'Back' ).click();
 		await page.getByRole( 'button', { name: 'Index' } ).click();
 
-		const editorFrame = page.frameLocator( 'iframe[name="editor-canvas"]' );
-		const headerTitle = editorFrame.getByRole( 'document', {
-			name: 'Block: Site Title',
-		} );
-		await expect( headerTitle ).toBeVisible();
-
-		await navigationSidebar.getByLabel( 'Back' ).click();
+		// Go back to the Custom Single Product template.
+		await page.getByLabel( 'Open Navigation' ).click();
 		await page
 			.getByRole( 'button', { name: 'Custom Single Product' } )
 			.click();
 
+		const editorFrame = page.frameLocator( 'iframe[name="editor-canvas"]' );
```

Comment:
Can we use `editor.canvas` here?

Reply:
Good catch! Updated in e15c594a21ff149034c2e322de62ceb652868782.